### PR TITLE
[Mirror][Localization] Translate forgotten phrases sentences #2254

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
These changes are a mirror of https://github.com/pagefaultgames/pokerogue/pull/2254

## Why am I doing these changes?
The owner of the previous PR @Nolway seems to have stopped working on it.
To continue previous work, I made a new PR and branch on EnochG1/mirror-translate-forgot-phrases

## What did change?
1. Move owners of the [translate-forgot-phrases](https://github.com/Nolway/pokerogue/tree/translate-forgot-phrases) branch
2. Resolve conflict with latest 'main' branch.
3. Remove duplicate message ("You received {{modifierName}}!" is already localized at https://github.com/pagefaultgames/pokerogue/pull/2556)

#### En
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

#### Fr
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

#### De
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

#### Pt_BR
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

#### Zh_CN
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

#### Ko
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

#### It
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| retryBattle | image |
| congratulations | image |
| unlockedSomething | image |
| beatModeFirstTime | image |

</details>

## How to test the changes?
Manually.
1. retryBattle : Make all pokémon fainted then check message.
2. unlockedSomething / congratulations / beatModeFirstTime : Clear Classic Mode with pokémon that have no history of completing Classic Mode. Using override.ts like this.
```Typescript
export const STARTING_WAVE_OVERRIDE: integer = 200;
export const STARTING_BIOME_OVERRIDE: Biome = Biome.END;
export const STARTER_FORM_OVERRIDES: Partial<Record<Species, number>> = {
  [Species.VENUSAUR]: 2
export const STARTING_LEVEL_OVERRIDE: integer = 500;
export const STARTER_SPECIES_OVERRIDE: Species | integer = Species.VENUSAUR;
};
```

## Checklist
- [ ] There is no overlap with another PR?
 - Only with https://github.com/pagefaultgames/pokerogue/pull/2254 which was mirrored
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/pt_BR/battle.ts [hunk 1]
```diff
@@ -77,6 +77,10 @@ export const battle: SimpleTranslationEntries = {
   "statSeverelyFell": "{{stats}} de {{pokemonNameWithAffix}} diminuiu severamente!",
   "statWontGoAnyLower": "{{stats}} de {{pokemonNameWithAffix}} não vai mais diminuir!",
   "ppReduced": "O PP do movimento {{moveName}} de\n{{targetName}} foi reduzido em {{reduction}}!",
+  "retryBattle": "Would you like to retry from the start of the battle?",
+  "unlockedSomething": "{{unlockedThing}}\nhas been unlocked.",
+  "congratulations": "Congratulations!",
+  "beatModeFirstTime": "{{speciesName}} beat {{gameMode}} Mode for the first time!\nYou received {{newModifier}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:83]:
```suggestion
  "retryBattle": "Você gostaria de tentar novamente desde o início da batalha?",
  "unlockedSomething": "{{unlockedThing}}\nfoi desbloqueado.",
  "congratulations": "Parabéns!",
  "beatModeFirstTime": "{{speciesName}} venceu o Modo {{gameMode}} pela primeira vez!\nVocê recebeu {{newModifier}}!",
```

**Comment by CodeTappert** [comment]
image
Why is it "mode" at the unlockable? This word then is not localized

File: src/locales/de/battle.ts [hunk 2]
```diff
@@ -77,6 +77,10 @@ export const battle: SimpleTranslationEntries = {
   "statSeverelyFell": "{{stats}} von {{pokemonNameWithAffix}} sinkt drastisch!",
   "statWontGoAnyLower": "{{stats}} von {{pokemonNameWithAffix}} kann nicht weiter sinken!",
   "ppReduced": "{{moveName}} von {{targetName}} wird um {{reduction}} AP reduziert!",
+  "retryBattle": "Möchtest du vom Beginn des Kampfes neustarten?",
+  "unlockedSomething": "{{unlockedThing}}\nwurde freigeschaltet.",
+  "congratulations": "Glückwunsch!",
+  "beatModeFirstTime": "{{speciesName}} hat den {{gameMode}} Modus zum ersten Mal beendet!\nDu erhältst {{newModifier}}!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:83]:
```suggestion
  "beatModeFirstTime": "{{speciesName}} hat den {{gameMode}} Modus zum ersten Mal beendet! Du erhältst {{newModifier}}!",
```

File: src/locales/de/battle.ts [hunk 3]
```diff
@@ -77,6 +77,10 @@ export const battle: SimpleTranslationEntries = {
   "statSeverelyFell": "{{stats}} von {{pokemonNameWithAffix}} sinkt drastisch!",
   "statWontGoAnyLower": "{{stats}} von {{pokemonNameWithAffix}} kann nicht weiter sinken!",
   "ppReduced": "{{moveName}} von {{targetName}} wird um {{reduction}} AP reduziert!",
+  "retryBattle": "Möchtest du vom Beginn des Kampfes neustarten?",
+  "unlockedSomething": "{{unlockedThing}}\nwurde freigeschaltet.",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:81]:
```suggestion
  "unlockedSomething": "{{unlockedThing}} wurde freigeschaltet.",
```

**Comment by EnochG1** [comment]
@CodeTappert 
unlockables.ts is not fully localized.
Maybe we should do it in a separate PR.

**Comment by EnochG1** [comment]
> Why is it "mode" at the unlockable? This word then is not localized

@CodeTappert 
unlockables.ts is not fully localized.
Maybe we should do it in a separate PR.

**Review by josericardo-fo — APPROVED** [review]
LGTM

**Comment by CodeTappert** [comment]
> > Why is it "mode" at the unlockable? This word then is not localized
> 
> @CodeTappert unlockables.ts is not fully localized. Maybe we should do it in a separate PR.

please do so

File: src/locales/zh_CN/battle.ts [hunk 4]
```diff
@@ -77,6 +77,10 @@ export const battle: SimpleTranslationEntries = {
   "statSeverelyFell": "{{pokemonNameWithAffix}}\n的{{stats}}极大幅降低了！",
   "statWontGoAnyLower": "{{pokemonNameWithAffix}}\n的{{stats}}已经无法再降低了！",
   "ppReduced": "降低了{{targetName}}的\n{{moveName}}的PP{{reduction}}点！",
+  "retryBattle": "你要从对战开始时重试么？",
+  "unlockedSomething": "{{unlockedThing}}\n已解锁。",
+  "congratulations": "恭喜！",
+  "beatModeFirstTime": "{{speciesName}} 首次击败了 {{gameMode}} 模式！\n你获得了{{newModifier}}！",
```
Comment (40chyan) [inline src/locales/zh_CN/battle.ts:83]:
```suggestion
  "beatModeFirstTime": "{{speciesName}}首次击败了{{gameMode}}！\n你获得了{{newModifier}}！",
```
we've already included "mode" in the replacement so I deleted “模式”

**Review by 40chyan — APPROVED** [review]

**Review by NicusPulcis — APPROVED** [review]
italian is ok

**Review by CodeTappert — APPROVED** [review]

**Review by Tempo-anon — APPROVED** [review]
Noting that es translation is missing

**Pull request merged**